Algorithms/Search/binarySearch.cp

```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+using namespace std;
+int binarySearch(int *arr,int n,int x){
+    int low=0,high=n-1,mid;
+    while(low<=high){        //we will stop running this loop when low will become greater than high
+        mid=low+(high-low)/2;
+        if(x==arr[mid]){   //if value of x become equal to mid value of array then we return mid
+            return mid;
+        }
+        else if(x<arr[mid]){     //if value of x is less than mid then high will become mid-1
+            high=mid-1;
+        }
+        else{     //if value of x is less than mid then high will become mid-1
+            low=mid+1;
+        }
+    }
+    return -1;       //if value is not in the array then we will return -1
+}
+int main() {
+    int n,x,ans;      //size of the array
+    cin>>n;
+    cin>>x;  //value of number
+    int arr[n+1];  //array
+    for(int i=0;i<n;i++){
+        cin>>arr[i];
+    }
+    ans=binarySearch(arr,n,x);
+    cout<<"The number "<<x<<" is found at position "<<ans+1<<"."<<endl;
+}
```

Comment:
Always return 0 at the end of `main()`

Reply:
Please fix your PR description, this is a new algorithm and not a bug fix or new data structure.

Reply:
Please remove extra files which is `.DS_Store` or add it to `.gitignore`

Reply:
all the changes are made @ay2306

Reply:
@Meghag5 you have committed some extra hidden files, probably set up by your text editor. Please delete them and add `.DS_Store` to `.gitignore` file.

Reply:
@ay2306 no such file is showing in my cloned file but it is showing on my github and when I am deleting it from github commits are increasing and after squashing it file is adding again.
So pls help how to resolve this problem

Reply:
Yes those are hidden file. Please note that you have added 4 files instead of 1. 3 hidden files are there.

Reply:
so pls help how can I delete it

Reply:
Okay follow these steps
1. On your Forked repo, delete those 3 hidden files (on GitHub only)
2. Add `.DS_Store` to `.gitignore` file.

It will be fine then.

Reply:
changes are made @ay2306